### PR TITLE
fix broadcasts with only adjoint/transpose SVectors

### PR DESCRIPTION
**Pull request by marius311**
A minor thing I ran into / fixed.

### CODE REVIEW

**Comment by mateuszbaran** [comment]
Thanks! I have a few comments:
  1. This change sometimes modifies the result type of broadcasts that mix different static arrays (for example `SArray` and `MArray`), for example in expressions like `S' .+ M` vs `S .+ M` where `S` is a `SArray` and `M` is an `MArray`. I don't know which result type is better, but the old behavior can be kept if necessary by only looking for `StaticArray`-like things only when no non-wrapped `StaticArray`s were found.
  2. You can use the `StaticArrayLike` union here instead of your custom one.

File: src/broadcast.jl [hunk 1]
```diff
@@ -97,7 +97,7 @@ scalar_getindex(x) = x
 scalar_getindex(x::Ref) = x[]
 
 @generated function _broadcast(f, ::Size{newsize}, s::Tuple{Vararg{Size}}, a...) where newsize
-    first_staticarray = a[findfirst(ai -> ai <: StaticArray, a)]
+    first_staticarray = a[findfirst(ai -> ai <: Union{StaticArray, Transpose{<:Any, <:StaticArray}, Adjoint{<:Any, <:StaticArray}}, a)]
```
Comment (tkoolen) [inline src/broadcast.jl]: Should we be using [`StaticArrayLike`](https://github.com/JuliaArrays/StaticArrays.jl/blob/103e9d4dabfdb64ccb378c8539d15136dde964de/src/StaticArrays.jl#L79-L98) here?

Comment (c42f) [reply src/broadcast.jl]: Yes ideally we'd add `StaticArrayLike` support to this method. But we'd need to make sure that `similar_type(first_staticarray, ...)` makes sense in all the cases that covers, and I think we'd need to add to `BroadcastStyle()` as well to intercept those wrapped arrays correctly.

Comment (marius311) [reply src/broadcast.jl]: Yea, the Union I chose was specifically to have it be consistent with the definitions for [BroadcastStyle](https://github.com/JuliaArrays/StaticArrays.jl/blob/b4f7e5e9bef5204f30275a86f61baba804fb1b64/src/broadcast.jl#L13-L15), so currently I think its at least consistent, if not exhaustive. 

Up to you guys if you want to merge this as-is or not.

**Comment by c42f** [comment]
I added a minor change to be clearer about what the test is actually testing.

I do think we should use `StaticArrayLike` more consistently but for now I'm happy to merge this as a bugfix rather than an enhancement (pending CI passing again of course).

**Comment by c42f** [comment]
>     1. This change sometimes modifies the result type of broadcasts

That's true... though I think it's easiest just to go with this change as I it follows the rule of "always use the leftmost StaticArray (Like)" with `similar_type`.

In general I feel that this existing leftmost rule isn't satisfying and we should probably do something better. But for now it has the advantage of being simple.

**Pull request merged**